### ISSUE TITLE
Possible tooth logger problem in October 2018 firmware

### ISSUE THREAD

**Issue by ric355** [noisymime/speeduino#190]
I think there might be a problem with the tooth logger in the October 2018 firmware. No matter what I try I can't seem to get it to generate any results. If I choose write to file, no file appears.  I do get sync during cranking so I'm pretty sure I have a crank signal.

I asked on Facebook and a couple of other people reported that they can't get it to work either.

**Comment by DeeEmm** [comment]
Do you use a Mac? My toothlogger has not been working since same update. I have not been able to solve this issue. I believe it may be an OSX related comms issue as others do not experience this on Windows. I did manage to establish that TS was opening up communications to the ECU but didn't get much further. There’s a thread in the forums about this. Should also note that downgrading did not fix the issue.

**Comment by ric355** [comment]
No I'm on Windows. I had several more replies on Facebook; one said it works for him on August but not on October. Another said that he did manage to get it working with October but only above 3000rpm. I can't go that high as I'm only cranking the motor.

**Comment by ric355** [comment]
I just saw someone post on the forum that tooth log doesn't work in cranking state, only running state? Is this correct?

I have my engine running again now, so will try it again at idle later on.

**Comment by noisymime** [comment]
I've just pushed a number of small fixes to the composite and tooth loggers to try and resolve some low speed issues (Amongst others). 

Are you able to test the current master code?

**Comment by ric355** [comment]
> I've just pushed a number of small fixes to the composite and tooth loggers to try and resolve some low speed issues (Amongst others).
> 
> Are you able to test the current master code?

As I was testing on the vehicle before and can't test the master firmware on the vehicle, I have repeated my test with October using Ardustim and then tested the master branch with Ardustim.

Results:
* October firmware - no results using ardustim, as expected.
* Master - results now appear in the log. I've tested at low cranking speed (100rpm), 100rpm and 6000rpm.

Ergo the problem is fixed!
